client/components/signup-form/index.jsx

```diff
@@ -311,13 +311,13 @@ export default React.createClass( {
 						autoCapitalize="off"
 						autoCorrect="off"
 						className="signup-form__input"
-						disabled={ this.state.submitting || this.props.disabled }
+						disabled={ this.state.submitting || this.props.disabled || ! this.props.allowEmailChange }
 						id="email"
 						name="email"
 						type="email"
 						value={ formState.getFieldValue( this.state.form, 'email' ) }
 						isError={ formState.isFieldInvalid( this.state.form, 'email' ) }
```

Comment:
I believe the answer is yes and no. 

If we leave it as is, we can end up with a form like this:

![match_email_logged_out_validation](https://cloud.githubusercontent.com/assets/1126811/12522639/60dd2f78-c117-11e5-9a51-50828945616b.png)

If we remove the validation, and then also do something like `! this.props.disableEmailInput` on [line 307](https://github.com/Automattic/wp-calypso/blob/update/invites-force-matching-email/client/components/signup-form/index.jsx#L307), we wouldn't have any validation for the email. So, the user could fill out the form, hit submit, and not have a clue why it's not working.

I think the best way to address this for now is to modify the error message that is returned. But, at this point, I'm fine doing that in a PR. I believe this case of forcing a matching email and then the invitee's email being taken is somewhat of an edge case.